test/mason/mason-test/mason-test.good

```diff
@@ -1,11 +1,17 @@
 Updating registry
 Downloading dependency: _MasonTest1-0.2.0
 
+======================================================================
+ERROR compileerror.chpl: compileerror
+----------------------------------------------------------------------
+compileerror failed to compile
 ======================================================================
 FAIL shallnotpass.chpl: shallnotpass
 ----------------------------------------------------------------------
 shallnotpass returned exitCode = 255
 ----------------------------------------------------------------------
-Ran 7 tests in n seconds
+Ran 8 tests in n seconds
 
-FAILED (passed = 6 failures = 1 )
+FAILED (passed = 6 failures = 1 errors = 1 )
+PATH/compileerror.chpl:1: syntax error: near 'test'
+compilation failed for compileerror.chpl
```

Comment:
Can you check why this message is getting printed?

Reply:
`PATH` is produced by prediff. We don't want the compiler to print the actual path, which causes trouble to compare.

Reply:
https://github.com/chapel-lang/chapel/blob/9ddf10c3e428566f335e2dc7f1995ee0a737c0d8/tools/mason/MasonTest.chpl#L196-L202
The error message is printed here. What I don't quite understand is why the message comes after the result. It should be vice versa.

Reply:
Tried adding `stdout.flash();stderr.flush();` following the `runWithStatus`. The output remains the same.
Manually running `mason test` in the directory emits output in the right order.

Reply:
I am not sure why this `PATH/compileerror.chpl:1: syntax error: near 'test'` is coming.

Reply:
As I mentioned above, `runWithStatus` will print the output from the subprocess, which is the compiler process in this case.

Reply:
Sorry my bad, I didn't notice the default parameterized value of `show`.
https://github.com/chapel-lang/chapel/blob/9ddf10c3e428566f335e2dc7f1995ee0a737c0d8/tools/mason/MasonUtils.chpl#L122

I suggest passing the show variable as well here.
https://github.com/chapel-lang/chapel/blob/9ddf10c3e428566f335e2dc7f1995ee0a737c0d8/tools/mason/MasonTest.chpl#L196
```chpl
const compilation = runWithStatus(compCommand, show);
```

Reply:
> Tried adding `stdout.flash();stderr.flush();` following the `runWithStatus`. The output remains the same.
> Manually running `mason test` in the directory emits output in the right order regardless of flushing or not.
> In spite of the strange order of output in `start_test`, it works well.

I am not very sure about it but this may be because of using pipe while printing. `start_test` might be printing the pipe value later after running the code. Try changing it to `writeln`

Reply:
> I suggest passing the show variable as well here.

Done.

> I am not very sure about it but this may be because of using pipe while printing. start_test might be printing the pipe value later after running the code. Try changing it to writeln

I suppose it's ok we leave that since it's more likely just some specific behavior we don't know about `start_test` for now. What do you think?

Reply:
So, even passing the `show` is not helping as the `stderr` pipe is not declared here.
https://github.com/chapel-lang/chapel/blob/9ddf10c3e428566f335e2dc7f1995ee0a737c0d8/tools/mason/MasonUtils.chpl#L126

Doing this solves the problem of printing the error. 
```chpl
var sub = spawn(cmd, stdout=PIPE, stderr=PIPE);
```

You can decide the way of printing when `show` is set.

Reply:
```
var sub = spawn(cmd, stdout=PIPE, stderr=PIPE);
```
makes the error message from the compiler disappear when running `start_test` on my machine.
```
 var sub = spawn(cmd, stdout=PIPE, stderr=STDOUT);
```
I also tried this. I think it's a reasonable change. However, no effect.

Reply:
Yes, I think we should not print the error message until the user is asking which can be done using `--show`.

@ben-albrecht any suggestions ?

Reply:
Well, in that case, `stderr=STDOUT` will work.

Reply:
Make sure all the tests are passing. I think after you latest commit `mason-test.good` file will also change.

Reply:
Actually I have had the same thought and tested it before https://github.com/chapel-lang/chapel/pull/15476/commits/bb79fc39361c3569057f29e8c6108e8cf3b4e6e9
To my surprise, it passed. I have mentioned it above.
 > However, no effect to the order of output.

Reply:
The following changes remove the error message.
```chapel
    var sub = spawn(cmd, stdout=PIPE, stderr=PIPE);

    var line:string;
    if show {
      while sub.stdout.readline(line) do write(line);
      while sub.stderr.readline(line) do write(line);
    }
```

Reply:
```bash
Updating registry
Downloading dependency: _MasonTest1-0.2.0

======================================================================
ERROR compileerror.chpl: compileerror
----------------------------------------------------------------------
compileerror failed to compile
======================================================================
FAIL shallnotpass.chpl: shallnotpass
----------------------------------------------------------------------
shallnotpass returned exitCode = 255
----------------------------------------------------------------------
Ran 8 tests in n seconds

FAILED (passed = 6 failures = 1 errors = 1 )
compilation failed for compileerror.chpl
```

This is the output I am getting.

Reply:
I have made the change.
Strangely enough https://github.com/chapel-lang/chapel/commit/bb79fc39361c3569057f29e8c6108e8cf3b4e6e9 does not work. The stderr should be redirected to its stdout and to the PIPE.
Another thing, I just found that I forgot about changing the UnitTest related code. I will find some time to get back on that and the redirection issue tomorrow.

Reply:
Also, many thanks for your patience to review this PR! :)